File: schemas/project-management/drizzle/task_attachments.ts
// task_attachments: files uploaded and linked to tasks.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, timestamp, index } from "drizzle-orm/pg-core";
import { tasks } from "./tasks";
import { users } from "./users";

export const taskAttachments = pgTable(
  "task_attachments",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    taskId: uuid("task_id").notNull().references(() => tasks.id, { onDelete: "cascade" }),
    uploadedBy: uuid("uploaded_by").references(() => users.id, { onDelete: "set null" }),
    fileName: text("file_name").notNull(),
    fileUrl: text("file_url").notNull(),
    fileSize: integer("file_size"),
    mimeType: text("mime_type"),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_task_attachments_task_id").on(table.taskId),
    index("idx_task_attachments_uploaded_by").on(table.uploadedBy),
  ]
);

File: schemas/project-management/spacetimedb/task_attachments.rs
// task_attachments: Files attached to tasks.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = task_attachments, public)]
pub struct TaskAttachment {
    #[primary_key]
    pub id: String, // UUID
    #[index(btree)]
    pub task_id: String, // UUID — FK → tasks.id (cascade delete)
    #[index(btree)]
    pub uploaded_by: Option<String>, // UUID — FK → users.id (set null)
    pub file_name: String,
    pub file_url: String,
    pub file_size: Option<i32>,
    pub mime_type: Option<String>,
    pub created_at: Timestamp,
}
